src/handlers/rollup-usage.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * Daily rollup Lambda function
+ * Queries CloudFront logs via Athena and stores aggregated usage in DynamoDB
+ */
+
+const {
+  AthenaClient,
+  StartQueryExecutionCommand,
+  GetQueryExecutionCommand,
+  GetQueryResultsCommand
+} = require('@aws-sdk/client-athena');
+const {
+  DynamoDBClient,
+  UpdateItemCommand
+} = require('@aws-sdk/client-dynamodb');
+
+const athena = new AthenaClient();
+const dynamodb = new DynamoDBClient();
+
+const ATHENA_DATABASE = process.env.ATHENA_DATABASE;
+const ATHENA_OUTPUT_BUCKET = process.env.ATHENA_OUTPUT_BUCKET;
+const USAGE_METRICS_TABLE = process.env.USAGE_METRICS_TABLE;
+
+/**
+ * Wait for Athena query to complete
+ */
+async function waitForQueryResults(queryExecutionId, maxAttempts = 60) {
+  for (let i = 0; i < maxAttempts; i++) {
+    const response = await athena.send(
+      new GetQueryExecutionCommand({ QueryExecutionId: queryExecutionId })
+    );
+
+    const state = response.QueryExecution.Status.State;
+
+    if (state === 'SUCCEEDED') {
+      return true;
+    } else if (state === 'FAILED' || state === 'CANCELLED') {
+      throw new Error(
+        `Query failed: ${response.QueryExecution.Status.StateChangeReason}`
+      );
+    }
+
+    // Wait 2 seconds before checking again
+    await new Promise(resolve => setTimeout(resolve, 2000));
+  }
+
+  throw new Error('Query timed out');
+}
+
+/**
+ * Execute Athena query and return results
+ */
+async function executeAthenaQuery(query) {
+  console.log('Executing Athena query:', query);
+
+  const queryExecution = await athena.send(
+    new StartQueryExecutionCommand({
+      QueryString: query,
+      QueryExecutionContext: { Database: ATHENA_DATABASE },
+      ResultConfiguration: { OutputLocation: ATHENA_OUTPUT_BUCKET },
+    })
+  );
+
+  const queryExecutionId = queryExecution.QueryExecutionId;
+  console.log('Query execution ID:', queryExecutionId);
+
+  await waitForQueryResults(queryExecutionId);
+
+  const results = await athena.send(
+    new GetQueryResultsCommand({ QueryExecutionId: queryExecutionId })
+  );
+
+  return results.ResultSet.Rows;
+}
+
+/**
+ * Parse API key from CloudFront query string
+ */
+function parseApiKeyFromQueryString(queryString) {
+  if (!queryString || queryString === '-') {
+    return 'no-key';
+  }
+
+  // Extract cf_api_key parameter that was added by CloudFront function
+  const match = queryString.match(/cf_api_key=([^&]+)/);
```

Comment:
Does the API key need to be parsed out of the query string or can we simply see it in the headers?